data/types/device/mode/mode.go

```diff
@@ -0,0 +1,79 @@
+package mode
+
+import (
+	"github.com/tidepool-org/platform/data"
+	commontypes "github.com/tidepool-org/platform/data/types/common"
+	"github.com/tidepool-org/platform/data/types/device"
+	"github.com/tidepool-org/platform/structure"
+)
+
+const (
+	ConfidentialMode = "confidential"
+	ZenMode          = "zen"
+)
+
+func Modes() []string {
+	return []string{
+		ConfidentialMode,
+		ZenMode,
+	}
+}
+
+type Mode struct {
+	device.Device `bson:",inline"`
+	EventID       *string               `json:"eventId,omitempty" bson:"eventId,omitempty"`
+	Duration      *commontypes.Duration `json:"duration,omitempty" bson:"duration,omitempty"`
+}
+
+func New(subType string) *Mode {
+	return &Mode{
+		Device: device.New(subType),
+	}
+}
+
+func NewWithEvent(subType string, deviceEvent string) *Mode {
+	return &Mode{
+		Device: device.NewWithEvent(subType, deviceEvent),
+	}
+}
+
+func (m *Mode) Parse(parser structure.ObjectParser) {
+	if !parser.HasMeta() {
+		parser = parser.WithMeta(m.Meta())
+	}
+
+	m.Device.Parse(parser)
+	m.EventID = parser.String("eventId")
+	m.Duration = commontypes.ParseDuration(parser.WithReferenceObjectParser("duration"))
+}
+
+func (m *Mode) Validate(validator structure.Validator) {
```

Comment:
EventType should be mandatory.
Since the verification that if it exists, it has a good value, we wan only verify that the string is not empty.

If not doing so, platform return an error 500 after a crash in data/types/device/factory.NewDeviceDatum()